build: introduce a package.bzl

alexeagle opened the pull request:
This lets Angular Bazel users install our transitive deps, rather than have to list them in their WORKSPACE file.
If they want a different version of one of these deps, they just need to install it before calling rules_angular_dependencies.

alexeagle committed:
build: introduce a package.bzl

This lets Angular Bazel users install our transitive deps, rather than have to list them in their WORKSPACE file.
If they want a different version of one of these deps, they just need to install it before calling rules_angular_dependencies.

mary-poppins commented:
You can preview e74ae6c at https://pr26125-e74ae6c.ngbuilds.io/.

mary-poppins commented:
You can preview dc3d4ee at https://pr26125-dc3d4ee.ngbuilds.io/.

gregmagolan commented:
rules_nodejs 0.14.0 will break the angular build for the time being until (due to BREAKING change of manual_build_file_contents attribute of yarn_install). https://github.com/bazelbuild/rules_typescript/pull/288 will resolve that.

mary-poppins commented:
You can preview c790161 at https://pr26125-c790161.ngbuilds.io/.

mary-poppins commented:
You can preview 42044aa at https://pr26125-42044aa.ngbuilds.io/.

alexeagle commented:
oh it was a mistake that this changed the versions of any deps. I only meant to move them to the other file.
Fixed, and also changed the approach so it mirrors the deps files from rules_nodejs and rules_ts, what do you think?

mary-poppins commented:
You can preview 81f5656 at https://pr26125-81f5656.ngbuilds.io/.

gregmagolan reviewed (approved):
Cool

gregmagolan commented:
Are the two files mirrored manually?

alexeagle commented:
Merged to `bazel` feature branch.
